src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -4202,12 +4202,24 @@ module TcDeclarations =
              | _ -> ()
         | ds ->
              // Check for duplicated parameters in abstract methods
+             // Check for an interface implementation with auto properties on constructor-less types
             for slot in ds do
                 if isAbstractSlot slot then
                     match slot with
                     | SynMemberDefn.AbstractSlot (slotSig = synVal; range = m) ->
                         CheckDuplicatesArgNames synVal m
                     | _ -> ()
+
+                if isInterface slot then
+                    match slot with
+                    | SynMemberDefn.Interface (members= Some defs) when not defs.IsEmpty ->
+                        let autoProps = defs |> List.filter(isAutoProperty)
```

Comment:
Do I get it right that it constructs a new list on each member, making the analysis quadratic?

Reply:
Sorry, looked at a wrong thing.

Reply:
I think a for loop and checking on the `SynMemberDefn.AutoProperty` would be better that filtering

Reply:
Happy to help!